### PR TITLE
add support devices and reformat (

### PR DESCRIPTION
**Pull request by G1K**
181a custom type pvvx
181b Xiaomi V2 Scale
181d Xiaomi V1 Scale
fe95 LYWSD02 and maybe mi flora
fee0 mi band steps, heartRate

add support auto discovery sensors to homeassistant
/write add custom payload type hex

and format indents

### CODE REVIEW

File: lib/attributes.js [hunk 1]
```diff
@@ -13,162 +13,283 @@
...
+    } else if ((a[0] & 0x0F) === 0x02) { // MKS kg
+      unit   = "kg";
+      weight = weight / 2;
+    } else {
+      unit = "???"
+    }
+    const state = {
+      isStabilized: ((a[1] & (1 << 5)) === 1),
+      loadRemoved: ((a[1] & (1 << 7)) === 1),
+      impedanceMeasured: ((a[1] & (1 << 1)) === 1)
```
Comment (gfwilliams) [inline lib/attributes.js]: I'm afraid this code is broken. These statements will *never* be true. The original code you had (`!== 0`) would have worked fine though.

**Comment by gfwilliams** [comment]
Thanks! This looks really exiting, especially with Home Assistant.

Do you think you could take a look at that code I mentioned above?

However, what is the reasoning behind reformatting all the Strings?

Stuff like this:

image

Just seems to make everything much less readable than it was before, and it makes it much harder for me to see if anything has changed that might have broken something.

File: lib/attributes.js (hunk 1 above)
Comment (G1K) [reply lib/attributes.js]: fixed

**Comment by G1K** [comment]
> Thanks! This looks really exiting, especially with Home Assistant.
> 
> Do you think you could take a look at that code I mentioned above?
> 
> However, what is the reasoning behind reformatting all the Strings?
> 
> Stuff like this:
> 
> image
> 
> Just seems to make everything much less readable than it was before, and it makes it much harder for me to see if anything has changed that might have broken something.


I wanted to stick to a single style of code and indentation. One style was adopted that changed the file the least, but in other files it seems to have slipped like this. Ideally, in the future, use eslint and configure the verification during PR . The only thing I can suggest in the current situation is to accept either single or double quotes, as you prefer.

**Comment by G1K** [comment]
PS: I just wanted to quickly format my and modified code, but without changing the rest, it didn't work. Formatting moved to a separate first commit https://github.com/espruino/EspruinoHub/pull/76/commits/c4632552b4eb85f77eea02def5e413c15f6c34e7

**Comment by gfwilliams** [comment]
Ok, thanks for the explanation. I might as well pull in as-is now, but in future I feel like it's better to avoid including code formatting in PRs with other stuff if at all possible. Your preferred code style is not my preferred style, and someone may come along in the future and just do another PR with *their* preferred style, then things get very messy!

I noticed a few other changes go in recently... Are you ready for me to merge, or shall I hold off?

**Comment by G1K** [comment]
Thank! Yes, this can already be merged, the current edits will be enough.
In the future, I would like to get the possibility of individual settings for devices, but how to do it better so as not to duplicate or break anything. This is necessary for an individual presence timeout and some other things, for example, the minimum signal strength or the list of filtered characteristics.

For the styling of the code, I'd like to see `.eslintrc.json` and stick to the common standard

**Comment by gfwilliams** [comment]
> how to do it better so as not to duplicate or break anything

What about something like:

```
"known_devices" : {
    "c0:52:3f:50:42:c9" : "original",
    "fc:a6:c6:04:db:79" : {
      "name" : "hall_down",
      "min_rssi" : -90
      "etc" : ...
    }
  },
```

**Pull request merged**